src/server/twemcache-rs/src/worker/single.rs

```diff
@@ -0,0 +1,288 @@
+    message_sender: Sender<Message>,
+    poll: Poll,
+    session_receiver: Receiver<Session>,
+    session_sender: Sender<Session>,
+    sessions: Slab<Session>,
+    data: SegCache<S>,
+}
+
+impl SingleWorker<CacheHasher> {
+    /// Create a new `Worker` which will get new `Session`s from the MPSC queue
+    pub fn new(config: Arc<Config>) -> Result<Self, std::io::Error> {
+        let poll = Poll::new().map_err(|e| {
+            error!("{}", e);
+            std::io::Error::new(std::io::ErrorKind::Other, "Failed to create epoll instance")
+        })?;
+        let sessions = Slab::<Session>::new();
+
+        let (session_sender, session_receiver) = crossbeam_channel::bounded(128);
+        let (message_sender, message_receiver) = crossbeam_channel::bounded(128);
+
+        let eviction = match config.segcache().eviction() {
+            Eviction::None => Policy::None,
+            Eviction::Random => Policy::Random,
+            Eviction::Fifo => Policy::Fifo,
+            Eviction::Cte => Policy::Cte,
+            Eviction::Util => Policy::Util,
+            Eviction::Merge => Policy::Merge {
+                max: config.segcache().merge_max(),
+                merge: config.segcache().merge_target(),
+                compact: config.segcache().compact_target(),
+            },
+        };
+
+        let data = SegCache::builder()
+            .power(config.segcache().hash_power())
+            .overflow_factor(config.segcache().overflow_factor())
+            .heap_size(config.segcache().heap_size())
+            .segment_size(config.segcache().segment_size())
+            .eviction(eviction)
+            .hasher(CacheHasher::default())
+            .build();
+
+        Ok(Self {
+            config,
+            poll,
+            message_receiver,
+            message_sender,
+            session_receiver,
+            session_sender,
+            sessions,
+            data,
+        })
+    }
+
+    /// Run the `Worker` in a loop, handling new session events
+    pub fn run(&mut self) {
+        let mut events = Events::with_capacity(self.config.worker().nevent());
+        let timeout = Some(std::time::Duration::from_millis(
+            self.config.worker().timeout() as u64,
+        ));
+
+        #[cfg(feature = "heap_dump")]
+        let mut ops = 0;
+        #[cfg(feature = "heap_dump")]
+        let mut seq = 0;
+
+        loop {
+            increment_counter!(&Stat::WorkerEventLoop);
+
+            self.data.expire();
+
+            // get events with timeout
+            if self.poll.poll(&mut events, timeout).is_err() {
+                error!("Error polling");
+            }
+
+            increment_counter_by!(
+                &Stat::WorkerEventTotal,
+                events.iter().count().try_into().unwrap(),
+            );
+
+            // process all events
+            for event in events.iter() {
+                let token = event.token();
+
+                // event for existing session
```

Comment:
I bet you know what I'm gonna say about the next 50 lines and some certain 50 lines in the MultiWorker 😛

Reply:
Should be fixed for both workers.